azavea/raster-foundry
# Send valid page parameters and catch invalid page parameters

## Overview

Updates the front-end to not send invalid page parameters (seen when clicking 'Try again' after a failed scene load).

Updates the back-end to ensure that `page.offset` is at least 0. Throws a 400 if it isn't, rather than allowing it to error with a 500.

### Notes

If this method of filtering for proper page parameters is acceptable, I'd like to go through the API and ensure all requests that use pagination are handled in the same way.


## Testing Instructions

 * Start the API server
 * Send a curl request with an invalid page parameter and ensure the response code is a 400
```
curl -s -o /dev/null -I -w "%{http_code}" \
     -H "Content-Type: application/json" \
     -H "Authorization: Bearer <TOKEN>" \
     http://localhost:9000/api/scenes?page=-1&pageSize=10&sort=createdAt,desc
```
 * Send a curl request with a valid page parameter and ensure the response code is a 200
```
curl -s -o /dev/null -I -w "%{http_code}" \
     -H "Content-Type: application/json" \
     -H "Authorization: Bearer <TOKEN>" \
     http://localhost:9000/api/scenes?page=0&pageSize=10&sort=createdAt,desc
```
 * Once the webpack server is running, access the `library/scenes/list` route and ensure the scenes are loaded
 * Log out and then reload the page
 * Log in (there should be an error message with a 'Try Again' link)
 * Click the 'Try Again' link and ensure the scene list is now loaded.

Resolves #705
